### PR TITLE
Closes d8-contrib-modules/akamai#16 - incorporating Akamai upstream libraries

### PR DESCRIPTION
**Pull request by cam8001**

### CODE REVIEW

**Comment by cam8001** [comment]
Getting close now. The real work of integrating Purge can start, calls to invalidate need to now update the Invalidation objects as Akamai clears them.

**Comment by nerdstein** [comment]
@cam8001 how is this going? I'd like to get the merge requests cleaned up

**Comment by nerdstein** [comment]
@cam8001 can you bring this home? It's been sitting for a while

File: akamai.services.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 services:
+  akamai.edgegridclient:
```
Comment (cam8001) [inline akamai.services.yml:2]: Not so sure about this name for the service. Might need a follow up issue to change it.

File: src/AkamaiContentControlClient.php [hunk 2]
```diff
@@ -51,17 +44,10 @@ class AkamaiContentControlClient implements AkamaiContentControlInterface {
    * @todo Inject HTTPClient
    */
   public function __construct(ConfigFactoryInterface $config_factory, LoggerInterface $logger) {
-    $this->config = $config_factory->get('akamai.settings');
     $this->logger = $logger;
-    $this->httpClient = \Drupal::httpClient();
+    $this->httpClient = AkamaiClient::create($config_factory);
```
Comment (cam8001) [inline src/AkamaiContentControlClient.php]: This whole file can go, it's no longer needed.

File: src/Form/AkamaiCacheControlForm.php [hunk 3]
```diff
@@ -70,15 +75,23 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    * {@inheritdoc}
    */
   public function submitForm(array &$form, FormStateInterface $form_state) {
-    $akamai = \Drupal::service('akamai.akamaiservice');
-    global $base_url;
-
+    $akamai = \Drupal::service('akamai.edgegridclient');
+    //$akamai = new AkamaiClient($this->configFactory());
```
Comment (cam8001) [inline src/Form/AkamaiCacheControlForm.php:79]: Remove

File: src/Form/AkamaiConfigForm.php [hunk 4]
```diff
@@ -50,11 +51,42 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       '#default_value' => $config->get('disable'),
     );
 
-    $form['akamai_restapi_endpoint'] = array(
+    // Akamai credentials
```
Comment (cam8001) [inline src/Form/AkamaiConfigForm.php]: Needs a better comment.

**Pull request merged**